src/command/trade.rs

```diff
@@ -516,17 +458,75 @@ async fn handle_finished_swap(
 
     maker.free_funds(dai, btc);
 
-    let _ = db
-        .remove_active_taker(&finished_swap.taker)
-        .await
-        .map_err(|error| tracing::error!("Unable to remove from active takers: {}", error));
-
     let _ = db
         .remove_swap(&swap_id)
         .await
         .map_err(|error| tracing::error!("Unable to delete swap from db: {}", error));
 }
 
+#[allow(clippy::too_many_arguments)]
+async fn handle_network_event(
+    network_event: network::Event,
+    maker: &mut Maker,
+    swarm: &mut Swarm,
+    db: Arc<Database>,
+    bitcoin_wallet: Arc<bitcoin::Wallet>,
+    ethereum_wallet: Arc<ethereum::Wallet>,
+    bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
+    ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
+    finished_swap_sender: Sender<FinishedSwap>,
+) {
+    match network_event {
+        network::Event::OrderMatch {
+            form,
+            to,
+            to_send,
+            common,
+            swap_protocol,
+            swap_id,
+        } => {
+            let result = maker.process_taken_order(form);
+
+            match result {
+                Ok(TakeRequestDecision::GoForSwap) => {
+                    if let Err(e) = swarm.setup_swap(&to, to_send, common, swap_protocol, swap_id) {
+                        tracing::error!("Sending setup swap message yielded error: {}", e)
+                    }
+                    // todo: publish new order here?
+                    // What if i publish a new order here and the does go through?
+                }
+                Ok(TakeRequestDecision::InsufficientFunds) => tracing::info!("Insufficient funds"),
+                Ok(TakeRequestDecision::RateNotProfitable) => tracing::info!("Rate not profitable"),
+                Err(e) => tracing::error!("Processing taken order yielded error: {}", e),
+            };
+        }
```

Comment:
I guess for now this is fine. In the long run though, the `TakeRequestDecision` has to go away. Nectar is not supposed to make a decision here whether or not it should move forward otherwise the whole idea of the orderbook is rendered invalid :sweat_smile:

Reply:
I will add a todo